lib/membership/index.js

```diff
@@ -230,6 +236,31 @@ Membership.prototype.makeSuspect = function makeSuspect(address, incarnationNumb
         Member.Status.suspect, this.localMember));
 };
 
+Membership.prototype.makeTombstone = function makeTombstone(address, incarnationNumber) {
+    this.ringpop.stat('increment', 'make-tombstone');
+    return this._updateMember(new Update(address, incarnationNumber,
+        Member.Status.tombstone, this.localMember));
+};
+
+Membership.prototype.removeMember = function removeMember(address) {
+    var hasMember = _.has(this.membersByAddress, address);
+    if (!hasMember) {
+        return;
+    }
+    var memberToDelete = this.membersByAddress[address];
+    delete this.membersByAddress[address];
+    this.members = _.without(this.members, memberToDelete);
+    this.computeChecksum();
+};
+
+Membership.prototype.evict = function evict(address) {
+    if (this.localMember.address === address) {
+        this.logger.error('ringpop tried to evict the local member from the memberlist, action has been prevented');
+        return;
+    }
+    this.removeMember(address);
```

Comment:
I have no prob. with moving it. I just emulated the Go implementation; as for why I originally made them separate in Go, I can't remember. Maybe I thought I'm going to reuse it somewhere.